driver-core/src/main/com/mongodb/ReadPreference.java

```diff
@@ -48,6 +48,45 @@
     ReadPreference() {
     }
 
+    /**
+     * Create a new ReadPreference instance with a new tag set.
+     *
+     * @param tagSet the new tag set
+     * @return a new ReadPreference instance with a new tag set
+     * @since 4.1
+     */
+    public abstract ReadPreference withTagSet(TagSet tagSet);
+
+    /**
+     * Create a new ReadPreference instance with a new tag set list.
+     *
+     * @param tagSet the new tag set list
+     * @return a new ReadPreference instance with a new tag set list
+     * @since 4.1
+     */
+    public abstract ReadPreference withTagSetList(List<TagSet> tagSet);
+
+    /**
+     * Create a new ReadPreference instance with the maximum acceptable staleness of a secondary in order to be considered for
+     * read operations.
+     *
+     * @param maxStalenessMS the max allowable staleness of secondaries. The minimum value is either 90 seconds, or the heartbeat frequency
+     *                       plus 10 seconds, whichever is greatest.
+     * @param timeUnit the time unit of maxStaleness
+     * @return a new ReadPreference instance with a new maximum allowable staleness
+     * @since 4.1
+     */
+    public abstract ReadPreference withMaxStalenessMS(Long maxStalenessMS, TimeUnit timeUnit);
+
+    /**
+     * Create a new ReadPreference instance where hedged reads are enabled/disabled in the server.
+     *
+     * @param hedgedReads true if hedged reads are enabled in the server
+     * @return a new ReadPreference instance where hedged reads are enabled/disabled in the server.
+     * @since 4.1
+     */
+    public abstract ReadPreference withHedgedReads(boolean hedgedReads);
```

Comment:
Let's make this method take a `ReadPreferenceHedgeOptions` instead of just a boolean, as we're likely to add more options in the future.